src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -279,6 +263,43 @@ void RadioSpinel<InterfaceType, ProcessContextType>::Init(bool aResetRadio,
     SuccessOrDie(error);
 }
 
+template <typename InterfaceType, typename ProcessContextType>
+void RadioSpinel<InterfaceType, ProcessContextType>::ResetRcp(bool aResetRadio)
+{
+    bool hardwareReset;
+    bool resetDone = false;
+
+    mIsReady    = false;
+    mWaitingKey = SPINEL_PROP_LAST_STATUS;
+
+    if (aResetRadio && (SendReset(SPINEL_RESET_STACK) == OT_ERROR_NONE) && (WaitResponse(false) == OT_ERROR_NONE))
+    {
+        otLogInfoPlat("Software reset RCP successfully");
+        ExitNow(resetDone = true);
+    }
+
+    hardwareReset = (mSpinelInterface.HardwareReset() == OT_ERROR_NONE);
+    VerifyOrExit(WaitResponse(false) == OT_ERROR_NONE);
+
+    resetDone = true;
+
+    if (hardwareReset)
+    {
+        otLogInfoPlat("Hardware reset RCP successfully");
+    }
+    else
+    {
+        otLogInfoPlat("RCP self reset successfully");
```

Comment:
Just curious, is this a common scenario?

Reply:
The simulation RCP handles this scenario.  The `no-reset` option can be set to `1` when running the simulation RCP (https://github.com/openthread/openthread/blob/main/tests/scripts/expect/posix-rcp-energy-scan.exp#L32). When the simulation RCP starts, it can guarantee that it has just reset itself.